src/account/operations/balance.rs

```diff
@@ -143,8 +173,19 @@ impl AccountHandle {
 
                                 // Add amount
                                 total_amount += amount;
+
                                 // Add storage deposit
-                                required_storage_deposit += output_data.output.rent_cost(&rent_structure);
+                                if output_data.output.is_basic() {
+                                    let rent = output_data.output.rent_cost(&rent_structure);
+
+                                    required_storage_deposit.basic += rent;
+                                } else if output_data.output.is_nft() {
+                                    let rent = output_data.output.rent_cost(&rent_structure);
+
+                                    required_storage_deposit.nft += rent;
+                                    total_rent_amount += rent;
+                                }
```

Comment:
How about changing this?
```suggestion
                                let rent = output_data.output.rent_cost(&rent_structure);
                                if output_data.output.is_basic() {
                                    required_storage_deposit.basic += rent;
                                    // amount for basic outputs isn't added to total_rent_amount, since we can also spent it without burning
                                } else if output_data.output.is_nft() {
                                    required_storage_deposit.nft += rent;
                                    total_rent_amount += rent;
                                }
```